tests/integration/steps/stargate_steps.go

```diff
@@ -0,0 +1,133 @@
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	resty "github.com/go-resty/resty/v2"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Stargate related functions
+func StargateService(t *testing.T, namespace string) (v1.Service, error) {
+	clientset, err := k8s.GetKubernetesClientFromOptionsE(t, getKubectlOptions(namespace))
+	g(t).Expect(err).To(BeNil())
+
+	services, _ := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{})
+	for _, service := range services.Items {
+		if strings.HasSuffix(service.ObjectMeta.Name, "-stargate-service") {
+			return service, nil
+		}
+	}
+	return v1.Service{}, fmt.Errorf("failed finding Stargate service")
+}
+
+func WaitForStargatePodReady(t *testing.T, namespace string) {
+	g(t).Eventually(func() bool {
+		return stargatePodIsReady(t, namespace)
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(BeTrue(), "Stargate deployment didn't roll out within timeout")
+}
+
+func stargatePodIsReady(t *testing.T, namespace string) bool {
+	output, err := k8s.RunKubectlAndGetOutputE(t, getKubectlOptions(namespace), "rollout", "status", "deployment", "k8ssandra-dc1-stargate")
+	if err == nil {
+		if strings.HasSuffix(output, "successfully rolled out") {
+			return true
+		}
+	}
+	return false
+}
+
+func WaitForAuthEndpoint(t *testing.T) {
+	g(t).Eventually(func() bool {
+		return authEndpointIsReachable(t)
+	}, 2*time.Minute, RETRY_INTERVAL).Should(BeTrue())
+}
+
+func authEndpointIsReachable(t *testing.T) bool {
+	restClient := resty.New()
+	response, err := restClient.R().Get("http://stargate.127.0.0.1.nip.io:8081/v1/auth")
+	if err != nil {
+		log.Printf("Failed connecting to Stargate auth endpoint: %s", err.Error())
+		return false
+	}
+	return response.StatusCode() == 405 // This endpoint should be invoked with a Post, we expect a 405 with a Get
```

Comment:
What if response is nil?  There are some other cases of this in some of the other functions as well.  I just didn't want us to get misleading errors that we have to trace if we need to protect them inside our verification functions.

Reply:
Response should only be `nil` if `err` was different than `nil` which means we've already exited at line 60. I think I've used pretty much the same pattern in different places indeed, but errors should be caught appropriately if needed.